### DISCUSSION TITLE
There is no current event loop in thread 'ScriptRunner.scriptThread' Error

### DISCUSSION THREAD

**Discussion by selcuktopal80** [fetchai/cosmpy#360] [Q&A]
i think asyncio.get_event_loop() method works within uagents module ?
and how to call asyncio.get_event_loop() function in uagents module?

class ASGIServer:
    def init(
        self,
        port: int,
        loop: asyncio.AbstractEventLoop,
        queries: Dict[str, asyncio.Future],
        logger: Optional[Logger] = None,
    ):
        self._port = int(port)
        self._loop = loop
        self._queries = queries
        self._logger = logger or get_logger("server")
        self._server = None
there is no get method so sadly 😦
because of that, it conflicts to streamlit module
and i get the following error:

File "C:\Users\selcu\AppData\Local\Programs\Python\Python310\lib\asyncio\events.py", line 656, in get_event_loop
    raise RuntimeError('There is no current event loop in thread %r.'
RuntimeError: There is no current event loop in thread 'ScriptRunner.scriptThread'.

**Comment by jrriehl** [comment]
Moved to uAgents issues: https://github.com/fetchai/uAgents/issues/101.
